src/util.cxx

```diff
@@ -9,14 +9,139 @@
 
 #include "util.h"
 
+#include <vector>
+#include <cctype>
+#include <cmath>
+
 NAMESPACE__THITTAM_UTIL__START
 
+
+std::string getString(const std::string &s, unsigned &pos) {
+  std::string returnValue = "";
+  while (isalpha(s[pos]))
+    {
+      returnValue += tolower(s[pos]);
+      pos++;
+    }
+  return returnValue;
+}
+
+
+double getNumber(const std::string &s, unsigned &pos) {
+  double x = 0.0;
+  bool sign = 0;
+
+  if (s[pos] == '-')
+    sign = true, pos++;
```

Comment:
Even if a block (if/for/while/) is only one line, you should include curly braces.